src/units/abilities.cpp

```diff
@@ -1229,6 +1229,40 @@ namespace { // Helpers for attack_type::special_active()
 
 }//anonymous namespace
 
+unit_ability_list attack_type::get_specials_and_abilities_with_priority(const std::string& special) const
+{
+	//the custom_priority attribute is used for define rules of priority of plague
+	unit_ability_list abil_list = get_specials_and_abilities(special);
+	unit_ability_list temp_list;
+	if(abil_list.empty()){
+		return abil_list;
+	}
+
+	std::vector<double> priorityvec;
+	for(const auto& i : abil_list) {
+		if((*i.ability_cfg)["priority"].to_double(0) >= 1){
+			priorityvec.push_back((*i.ability_cfg)["priority"].to_double(0));
+		}
+	}
+
+	if(!priorityvec.empty()){
+		double priority = *(std::min_element(priorityvec.begin(), priorityvec.end()));
+		int prc = std::count(priorityvec.begin(), priorityvec.end(), priority);
+		for(const auto& i : abil_list) {
+			if((*i.ability_cfg)["priority"].to_double(0) == priority) {
+				temp_list.emplace_back(i);
+				if(prc == 1){
```

Comment:
I don't understand what this is doing. I think `prc` is meant to optimise the loop when there's only one ability with the top priority, and the `return temp_list` returns the same result as the later `if(!temp_list.empty())` conditional would do without this optimisation.

However, calculating `prc` requires an extra scan of the list, so it isn't an optimisation.

Reply:
removed